lms/static/scripts/frontend_apps/components/ErrorDialogApp.js

```diff
@@ -0,0 +1,65 @@
+import { Fragment, createElement } from 'preact';
+import { useContext } from 'preact/hooks';
+
+import { Config } from '../config';
+
+import ErrorDisplay from './ErrorDisplay';
+import Dialog from './Dialog';
+
+/**
+ * Error dialog displayed when authorization with a third-party API via OAuth
+ * fails.
+ *
+ */
+export default function ErrorDialogApp() {
+  const {
+    ErrorDialog: { errorCode, errorDetails = '' },
+  } = useContext(Config);
+
+  const error = { code: errorCode, details: errorDetails };
+
+  const title = (() => {
+    return (
+      {
+        reused_tool_guid:
+          'The Hypothesis credentials your install is using may have already been used in a different LMS instance',
+      }[errorCode] || 'An error occurred'
```

Comment:
The default here are not supposed to happen, but, just in case, not sure if the non descriptive error dialog is better than letting  the exception bubble up to sentry or similar.